hbase-common/src/main/java/org/apache/hadoop/hbase/io/crypto/Encryption.java

```diff
@@ -215,7 +215,7 @@ public static boolean failOnHashAlgorithmMismatch(Configuration conf) {
    * Return the MD5 digest of the concatenation of the supplied arguments.
    */
   public static byte[] hash128(String... args) {
-    return hashWithAlg("MD5", args);
+    return hashWithAlg("MD5", Bytes.toByteArrays(args));
```

Comment:
hashWithAlg() already does Bytes.toBytes(), do we still need do this?

Reply:
so hashWithAlg takes any number of byte arrays, and don't use Bytes.toBytes(). 
(Bytes.toBytes was used e.g. by the old `hash128(String... args)` function I refactored)

So in my code: 
```
 public static byte[] hashWithAlg(String algorithm, byte[]... args) {
    try {
      MessageDigest md = MessageDigest.getInstance(algorithm);
      for (byte[] arg: args) {
        md.update(arg);
      }
      return md.digest();
    } catch (NoSuchAlgorithmException e) {
      throw new RuntimeException("unable to use hash algorithm: " + algorithm, e);
    }
  }

```

And here in `hash128` we are using the `Bytes.toByteArrays` function to convert the `String... args` to `byte[][]`.

Reply:
the method `hashWithAlg` needs a vararg array of byte arrays to operate on. This call here is coercing the vararg array of Strings into an array of byte arrays. AFAIK it's needed.

Reply:
It's fine. hashWithAlg is private anyways so there shouldn't be any need to pass a String argument ever.